### PR TITLE
Add error handling on backend and validation over boundary model

### PR DESCRIPTION
**Pull request by moradology**
This changeset is here to allow for more sophisticated frontend representations of failure.

### CODE REVIEW

File: ashlar/views.py [hunk 1]
```diff
@@ -71,6 +73,16 @@ class BoundaryViewSet(viewsets.ModelViewSet):
     serializer_class = BoundarySerializer
     filter_class = BoundaryFilter
 
+    def create(self, request, *args, **kwargs):
```
Comment (CloudNiner) [inline ashlar/views.py:76]: Not sure I understand why this is here ATM. There is no way this would fail, since uniqueness is on an auto generated primary key for this field. (And I kind of assumed the rest framework would throw this error already for uniqueness constraint failures?)

File: ashlar/views.py [hunk 2]
```diff
@@ -71,6 +73,16 @@ class BoundaryViewSet(viewsets.ModelViewSet):
...
+        try:
+            super(BoundaryViewSet, self).create(request, *args, **kwargs)
+        except IntegrityError:
+            return Response({'error': 'uniqueness constraint violation'}, status.HTTP_409_CONFLICT)
+
+    def perform_create(self, serializer):
```
Comment (CloudNiner) [inline ashlar/views.py]: Docstring. Why is this here?

File: ashlar/views.py (hunk 1 above)
Comment (moradology) [inline ashlar/views.py:76]: Prior to this addition, 500s were getting thrown, which made parsing output on the front end a bit more difficult.

File: ashlar/views.py (hunk 2 above)
Comment (moradology) [inline ashlar/views.py]: For these file uploads; creation requires only label and source_file - modifying perform_create allows us to create our instance with the uploaded file. 500s abound otherwise.

**Comment by CloudNiner** [comment]
+1 on tests passing
